src/wallet/rpcwallet.cpp

```diff
@@ -2591,9 +2581,7 @@ static UniValue getwalletinfo(const JSONRPCRequest& request)
     obj.pushKV("timefirstkey", pwallet->GetTimeFirstKey());
     obj.pushKV("keypoololdest", pwallet->GetOldestKeyPoolTime());
     obj.pushKV("keypoolsize",   (int64_t)pwallet->KeypoolCountExternalKeys());
-    if (fHDEnabled) {
-        obj.pushKV("keypoolsize_hd_internal",   (int64_t)(pwallet->KeypoolCountInternalKeys()));
-    }
+    obj.pushKV("keypoolsize_hd_internal",   (int64_t)(pwallet->KeypoolCountInternalKeys()));
```

Comment:
why'd we get rid of this check?

Reply:
With 14075 we can use `importmulti` to import pubkeys via `desc` option into _internal_ keypool even for non-HD/empty wallets. Another option would be to can go with `if (pwallet->CanSupportFeature(FEATURE_HD)) {` (like in btc) here and require users to upgrade their wallets for this to work correctly.